cli.js

```diff
@@ -51,15 +61,14 @@ if (cmd === 'init') {
 
 	const name = argv._[1]
 	if ('string' !== typeof name || !name) showError('Missing name.')
-	// todo: use get-port if no port given
-	const port = parseInt(argv._[2])
-	if (Number.isNaN(port) || port <= 0) showError('Invalid or missing port.')
 
-	const id = randomBytes(8).toString('hex')
+	ensurePort(parseInt(argv._[2])).then(port => {
+		const id = randomBytes(8).toString('hex')
 
-	mkdirp.sync(path.dirname(cfgPath))
-	fs.writeFileSync(cfgPath, JSON.stringify({id, name, port}))
-	console.info('Done.')
+		mkdirp.sync(path.dirname(cfgPath))
+		fs.writeFileSync(cfgPath, JSON.stringify({ id, name, port }))
```

Comment:
As above.

Reply:
Do you mean that the call to fs.writeFileSync should not be used there either? Since this call needs to be inside of the callback because it needs the port number, how do you feel about using the asynchronous variant of that method instead (fs.writeFile)?

Reply:
Yup, `fs.writeFile` + `new Promise(…)` or `util.promisify(fs.writeFile)` would work.

Reply:
I went with util.promisify